### PR TITLE
[TRAFFIC-9343][TRAFFIC-9344] Add confluent network network-link endpoint list/describe 

### PR DESCRIPTION
**Pull request by ay-zhang**
What
----

Note that this is merging into the feature branch https://github.com/confluentinc/cli/tree/traffic-8850-network-cli-az. We will merge commits into main from this feature branch after all of the commands for network are implemented.

- Add `confluent network network-link endpoint list`
- Add `confluent network network-link endpoint describe`

References
----------

[CLI Commands Design](https://confluentinc.atlassian.net/wiki/spaces/PM/pages/2987394300/1-pager+--+Network+CLI)
[Implementation 1-Pager](https://confluentinc.atlassian.net/wiki/spaces/~712020463029eabbac401bbe1b033ce0d3a00e/pages/3136749631/Network+CLI)

Test & Review
-------------

Added integration tests.

Performed manual tests.

### CODE REVIEW

File: internal/network/command_network_link_endpoint_list.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		}
+		if network != "" && endpoint.Spec.Network.GetId() != network {
+			continue
+		}
+		if service != "" && endpoint.Spec.NetworkLinkService.GetId() != service {
+			continue
+		}
+		if phase != "" && endpoint.Status.GetPhase() != phase {
+			continue
+		}
```
Comment (syoung-cflt) [inline internal/network/command_network_link_endpoint_list.go:87]: I assume you are trying to filter the endpoints based on the flags?

The list API actually supports filtering with query parameters. There was an issue for `MultipleSearchFilter` in the SDK so I didn't add filtering for our LIST commands.

Comment (syoung-cflt) [reply internal/network/command_network_link_endpoint_list.go:87]: https://docs.confluent.io/cloud/current/api.html#tag/Network-Link-Endpoints-(networkingv1)/operation/listNetworkingV1NetworkLinkEndpoints

1. You would need to update the SDK version to ([networking/v0.9.0](https://github.com/confluentinc/ccloud-sdk-go-v2/releases/tag/networking%2Fv0.9.0) and [networking-ip/v0.2.0](https://github.com/confluentinc/ccloud-sdk-go-v2/releases/tag/networking-ip%2Fv0.2.0))
2. Update the flags based on the API Spec (some of them should be `[]string`)
3. Update the `getNetworkLinkEndpoints` function (see [ref](https://github.com/confluentinc/ccloud-sdk-go-v2/blob/master/networking/v1/docs/NetworkLinkEndpointsNetworkingV1Api.md#listnetworkingv1networklinkendpoints ) to learn how to use the filters in the SDK)

Comment (ay-zhang) [reply internal/network/command_network_link_endpoint_list.go]: In [networking/v0.9.0](https://github.com/confluentinc/ccloud-sdk-go-v2/releases/tag/networking%2Fv0.9.0), the workaround for `MultipleSearchFilter` resulted in a few models to use `Set` type. Of the updated models,`ZonesInfo` is not compatible with type `Set`. I've removed flags here, will add them [separately](https://confluentinc.atlassian.net/browse/TRAFFIC-9348) once SDK is [updated](https://confluent.slack.com/archives/C04PBPC4H1T/p1701712403845029). CC @syoung-cflt @ejagade

File: internal/network/command_network_link_endpoint_list.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+)
+
+type listNetworkLinkEndpointOut struct {
+	Id                 string `human:"ID" serialized:"id"`
+	Name               string `human:"Name" serialized:"name"`
+	Network            string `human:"Network" serialized:"network"`
+	Description        string `human:"Description,omitempty" serialized:"description,omitempty"`
+	NetworkLinkService string `human:"Network Link Service" serialized:"network_link_service"`
+	Phase              string `human:"Phase" serialized:"phase"`
+}
```
Comment (brianstrauch) [inline internal/network/command_network_link_endpoint_list.go:20]: Looks like this is the same struct as `describe`, just without the "Environment" field. You can use the `Filter()` function so you don't need to redeclare a new struct.

Comment (ay-zhang) [reply internal/network/command_network_link_endpoint_list.go]: Updated!

File: internal/network/command_network_link_endpoint_list.go [hunk 3]
```diff
@@ -49,14 +40,16 @@ func (c *command) networkLinkEndpointList(cmd *cobra.Command, _ []string) error
 			return fmt.Errorf(errors.CorruptedNetworkResponseErrorMsg, "status")
 		}
 
-		list.Add(&listNetworkLinkEndpointOut{
+		list.Add(&networkLinkEndpointOut{
 			Id:                 endpoint.GetId(),
 			Name:               endpoint.Spec.GetDisplayName(),
 			Network:            endpoint.Spec.Network.GetId(),
-			NetworkLinkService: endpoint.Spec.NetworkLinkService.GetId(),
+			Environment:        endpoint.Spec.Environment.GetId(),
```
Comment (brianstrauch) [inline internal/network/command_network_link_endpoint_list.go:47]: This can be removed since it will be filtered out anyways
```suggestion
```

**Review by syoung-cflt — APPROVED** [review]

**Pull request merged**